// @arduino/arduino-iot-client/dist/model/ArduinoSeriesRawResponse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _BatchQueryRawResponseSeriesMediaV = _interopRequireDefault(require("./BatchQueryRawResponseSeriesMediaV1"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ArduinoSeriesRawResponse model module.
 * @module model/ArduinoSeriesRawResponse
 * @version 1.4.2
 */
var ArduinoSeriesRawResponse =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>ArduinoSeriesRawResponse</code>.
   * ArduinoSeriesRawResponse media type (default view)
   * @alias module:model/ArduinoSeriesRawResponse
   * @param countValues {Number} Total number of values in the array 'values'
   * @param fromDate {Date} From date
   * @param query {String} Query of for the data
   * @param respVersion {Number} Response version
   * @param series {module:model/BatchQueryRawResponseSeriesMediaV1} 
   * @param sort {module:model/ArduinoSeriesRawResponse.SortEnum} Sorting
   * @param status {String} Status of the response
   * @param times {Array.<Date>} Timestamp in RFC3339
   * @param toDate {Date} To date
   * @param values {Array.<Object>} Values can be in Float, String, Bool, Object
   */
  function ArduinoSeriesRawResponse(countValues, fromDate, query, respVersion, series, sort, status, times, toDate, values) {
    _classCallCheck(this, ArduinoSeriesRawResponse);

    ArduinoSeriesRawResponse.initialize(this, countValues, fromDate, query, respVersion, series, sort, status, times, toDate, values);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ArduinoSeriesRawResponse, null, [{
    key: "initialize",
    value: function initialize(obj, countValues, fromDate, query, respVersion, series, sort, status, times, toDate, values) {
      obj['count_values'] = countValues;
      obj['from_date'] = fromDate;
      obj['query'] = query;
      obj['resp_version'] = respVersion;
      obj['series'] = series;
      obj['sort'] = sort;
      obj['status'] = status;
      obj['times'] = times;
      obj['to_date'] = toDate;
      obj['values'] = values;
    }
    /**
     * Constructs a <code>ArduinoSeriesRawResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoSeriesRawResponse} obj Optional instance to populate.
     * @return {module:model/ArduinoSeriesRawResponse} The populated <code>ArduinoSeriesRawResponse</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoSeriesRawResponse();

        if (data.hasOwnProperty('count_values')) {
          obj['count_values'] = _ApiClient["default"].convertToType(data['count_values'], 'Number');
        }

        if (data.hasOwnProperty('from_date')) {
          obj['from_date'] = _ApiClient["default"].convertToType(data['from_date'], 'Date');
        }

        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }

        if (data.hasOwnProperty('query')) {
          obj['query'] = _ApiClient["default"].convertToType(data['query'], 'String');
        }

        if (data.hasOwnProperty('resp_version')) {
          obj['resp_version'] = _ApiClient["default"].convertToType(data['resp_version'], 'Number');
        }

        if (data.hasOwnProperty('series')) {
          obj['series'] = _BatchQueryRawResponseSeriesMediaV["default"].constructFromObject(data['series']);
        }

        if (data.hasOwnProperty('series_limit')) {
          obj['series_limit'] = _ApiClient["default"].convertToType(data['series_limit'], 'Number');
        }

        if (data.hasOwnProperty('sort')) {
          obj['sort'] = _ApiClient["default"].convertToType(data['sort'], 'String');
        }

        if (data.hasOwnProperty('status')) {
          obj['status'] = _ApiClient["default"].convertToType(data['status'], 'String');
        }

        if (data.hasOwnProperty('times')) {
          obj['times'] = _ApiClient["default"].convertToType(data['times'], ['Date']);
        }

        if (data.hasOwnProperty('to_date')) {
          obj['to_date'] = _ApiClient["default"].convertToType(data['to_date'], 'Date');
        }

        if (data.hasOwnProperty('values')) {
          obj['values'] = _ApiClient["default"].convertToType(data['values'], [Object]);
        }
      }

      return obj;
    }
  }]);

  return ArduinoSeriesRawResponse;
}();
/**
 * Total number of values in the array 'values'
 * @member {Number} count_values
 */


ArduinoSeriesRawResponse.prototype['count_values'] = undefined;
/**
 * From date
 * @member {Date} from_date
 */

ArduinoSeriesRawResponse.prototype['from_date'] = undefined;
/**
 * If the response is different than 'ok'
 * @member {String} message
 * @default ''
 */

ArduinoSeriesRawResponse.prototype['message'] = '';
/**
 * Query of for the data
 * @member {String} query
 */

ArduinoSeriesRawResponse.prototype['query'] = undefined;
/**
 * Response version
 * @member {Number} resp_version
 */

ArduinoSeriesRawResponse.prototype['resp_version'] = undefined;
/**
 * @member {module:model/BatchQueryRawResponseSeriesMediaV1} series
 */

ArduinoSeriesRawResponse.prototype['series'] = undefined;
/**
 * Max of values
 * @member {Number} series_limit
 */

ArduinoSeriesRawResponse.prototype['series_limit'] = undefined;
/**
 * Sorting
 * @member {module:model/ArduinoSeriesRawResponse.SortEnum} sort
 */

ArduinoSeriesRawResponse.prototype['sort'] = undefined;
/**
 * Status of the response
 * @member {String} status
 */

ArduinoSeriesRawResponse.prototype['status'] = undefined;
/**
 * Timestamp in RFC3339
 * @member {Array.<Date>} times
 */

ArduinoSeriesRawResponse.prototype['times'] = undefined;
/**
 * To date
 * @member {Date} to_date
 */

ArduinoSeriesRawResponse.prototype['to_date'] = undefined;
/**
 * Values can be in Float, String, Bool, Object
 * @member {Array.<Object>} values
 */

ArduinoSeriesRawResponse.prototype['values'] = undefined;
/**
 * Allowed values for the <code>sort</code> property.
 * @enum {String}
 * @readonly
 */

ArduinoSeriesRawResponse['SortEnum'] = {
  /**
   * value: "ASC"
   * @const
   */
  "ASC": "ASC",

  /**
   * value: "DESC"
   * @const
   */
  "DESC": "DESC"
};
var _default = ArduinoSeriesRawResponse;
exports["default"] = _default;
