// @arduino/arduino-iot-client/dist/model/ArduinoDevicev2propertyvalueValueStatistics.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ArduinoDevicev2propertyvalueValueStatistics model module.
 * @module model/ArduinoDevicev2propertyvalueValueStatistics
 * @version 1.4.2
 */
var ArduinoDevicev2propertyvalueValueStatistics =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>ArduinoDevicev2propertyvalueValueStatistics</code>.
   * @alias module:model/ArduinoDevicev2propertyvalueValueStatistics
   */
  function ArduinoDevicev2propertyvalueValueStatistics() {
    _classCallCheck(this, ArduinoDevicev2propertyvalueValueStatistics);

    ArduinoDevicev2propertyvalueValueStatistics.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ArduinoDevicev2propertyvalueValueStatistics, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ArduinoDevicev2propertyvalueValueStatistics</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoDevicev2propertyvalueValueStatistics} obj Optional instance to populate.
     * @return {module:model/ArduinoDevicev2propertyvalueValueStatistics} The populated <code>ArduinoDevicev2propertyvalueValueStatistics</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoDevicev2propertyvalueValueStatistics();

        if (data.hasOwnProperty('adr')) {
          obj['adr'] = _ApiClient["default"].convertToType(data['adr'], 'Number');
        }

        if (data.hasOwnProperty('channel')) {
          obj['channel'] = _ApiClient["default"].convertToType(data['channel'], 'Number');
        }

        if (data.hasOwnProperty('duplicate')) {
          obj['duplicate'] = _ApiClient["default"].convertToType(data['duplicate'], 'Number');
        }

        if (data.hasOwnProperty('freq')) {
          obj['freq'] = _ApiClient["default"].convertToType(data['freq'], 'Number');
        }

        if (data.hasOwnProperty('modBW')) {
          obj['modBW'] = _ApiClient["default"].convertToType(data['modBW'], 'Number');
        }

        if (data.hasOwnProperty('rssi')) {
          obj['rssi'] = _ApiClient["default"].convertToType(data['rssi'], 'Number');
        }

        if (data.hasOwnProperty('seqno')) {
          obj['seqno'] = _ApiClient["default"].convertToType(data['seqno'], 'Number');
        }

        if (data.hasOwnProperty('sf')) {
          obj['sf'] = _ApiClient["default"].convertToType(data['sf'], 'Number');
        }

        if (data.hasOwnProperty('snr')) {
          obj['snr'] = _ApiClient["default"].convertToType(data['snr'], 'Number');
        }

        if (data.hasOwnProperty('time')) {
          obj['time'] = _ApiClient["default"].convertToType(data['time'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return ArduinoDevicev2propertyvalueValueStatistics;
}();
/**
 * @member {Number} adr
 */


ArduinoDevicev2propertyvalueValueStatistics.prototype['adr'] = undefined;
/**
 * @member {Number} channel
 */

ArduinoDevicev2propertyvalueValueStatistics.prototype['channel'] = undefined;
/**
 * @member {Number} duplicate
 */

ArduinoDevicev2propertyvalueValueStatistics.prototype['duplicate'] = undefined;
/**
 * @member {Number} freq
 */

ArduinoDevicev2propertyvalueValueStatistics.prototype['freq'] = undefined;
/**
 * @member {Number} modBW
 */

ArduinoDevicev2propertyvalueValueStatistics.prototype['modBW'] = undefined;
/**
 * @member {Number} rssi
 */

ArduinoDevicev2propertyvalueValueStatistics.prototype['rssi'] = undefined;
/**
 * @member {Number} seqno
 */

ArduinoDevicev2propertyvalueValueStatistics.prototype['seqno'] = undefined;
/**
 * @member {Number} sf
 */

ArduinoDevicev2propertyvalueValueStatistics.prototype['sf'] = undefined;
/**
 * @member {Number} snr
 */

ArduinoDevicev2propertyvalueValueStatistics.prototype['snr'] = undefined;
/**
 * @member {Number} time
 */

ArduinoDevicev2propertyvalueValueStatistics.prototype['time'] = undefined;
var _default = ArduinoDevicev2propertyvalueValueStatistics;
exports["default"] = _default;
