// @arduino/arduino-iot-client/dist/model/ArduinoLorafreqplanv1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ArduinoLorafreqplanv1 model module.
 * @module model/ArduinoLorafreqplanv1
 * @version 1.4.2
 */
var ArduinoLorafreqplanv1 =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>ArduinoLorafreqplanv1</code>.
   * ArduinoLorafreqplanv1 media type (default view)
   * @alias module:model/ArduinoLorafreqplanv1
   * @param advanced {Boolean} Frequency plan only for advanced users
   * @param id {String} The ID of the frequency paln
   * @param name {String} The name of the frequency plan
   */
  function ArduinoLorafreqplanv1(advanced, id, name) {
    _classCallCheck(this, ArduinoLorafreqplanv1);

    ArduinoLorafreqplanv1.initialize(this, advanced, id, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ArduinoLorafreqplanv1, null, [{
    key: "initialize",
    value: function initialize(obj, advanced, id, name) {
      obj['advanced'] = advanced;
      obj['id'] = id;
      obj['name'] = name;
    }
    /**
     * Constructs a <code>ArduinoLorafreqplanv1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoLorafreqplanv1} obj Optional instance to populate.
     * @return {module:model/ArduinoLorafreqplanv1} The populated <code>ArduinoLorafreqplanv1</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoLorafreqplanv1();

        if (data.hasOwnProperty('advanced')) {
          obj['advanced'] = _ApiClient["default"].convertToType(data['advanced'], 'Boolean');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ArduinoLorafreqplanv1;
}();
/**
 * Frequency plan only for advanced users
 * @member {Boolean} advanced
 */


ArduinoLorafreqplanv1.prototype['advanced'] = undefined;
/**
 * The ID of the frequency paln
 * @member {String} id
 */

ArduinoLorafreqplanv1.prototype['id'] = undefined;
/**
 * The name of the frequency plan
 * @member {String} name
 */

ArduinoLorafreqplanv1.prototype['name'] = undefined;
var _default = ArduinoLorafreqplanv1;
exports["default"] = _default;
