// @arduino/arduino-iot-client/dist/model/ArduinoThing.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ArduinoProperty = _interopRequireDefault(require("./ArduinoProperty"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ArduinoThing model module.
 * @module model/ArduinoThing
 * @version 1.4.2
 */
var ArduinoThing =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>ArduinoThing</code>.
   * ArduinoThing media type (default view)
   * @alias module:model/ArduinoThing
   * @param href {String} The api reference of this thing
   * @param id {String} The id of the thing
   * @param name {String} The friendly name of the thing
   * @param timezone {String} Time zone of the thing
   * @param userId {String} The user id of the owner
   */
  function ArduinoThing(href, id, name, timezone, userId) {
    _classCallCheck(this, ArduinoThing);

    ArduinoThing.initialize(this, href, id, name, timezone, userId);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ArduinoThing, null, [{
    key: "initialize",
    value: function initialize(obj, href, id, name, timezone, userId) {
      obj['href'] = href;
      obj['id'] = id;
      obj['name'] = name;
      obj['timezone'] = timezone;
      obj['user_id'] = userId;
    }
    /**
     * Constructs a <code>ArduinoThing</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoThing} obj Optional instance to populate.
     * @return {module:model/ArduinoThing} The populated <code>ArduinoThing</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoThing();

        if (data.hasOwnProperty('created_at')) {
          obj['created_at'] = _ApiClient["default"].convertToType(data['created_at'], 'Date');
        }

        if (data.hasOwnProperty('deleted_at')) {
          obj['deleted_at'] = _ApiClient["default"].convertToType(data['deleted_at'], 'Date');
        }

        if (data.hasOwnProperty('device_fqbn')) {
          obj['device_fqbn'] = _ApiClient["default"].convertToType(data['device_fqbn'], 'String');
        }

        if (data.hasOwnProperty('device_id')) {
          obj['device_id'] = _ApiClient["default"].convertToType(data['device_id'], 'String');
        }

        if (data.hasOwnProperty('device_name')) {
          obj['device_name'] = _ApiClient["default"].convertToType(data['device_name'], 'String');
        }

        if (data.hasOwnProperty('device_type')) {
          obj['device_type'] = _ApiClient["default"].convertToType(data['device_type'], 'String');
        }

        if (data.hasOwnProperty('href')) {
          obj['href'] = _ApiClient["default"].convertToType(data['href'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('organization_id')) {
          obj['organization_id'] = _ApiClient["default"].convertToType(data['organization_id'], 'String');
        }

        if (data.hasOwnProperty('properties')) {
          obj['properties'] = _ApiClient["default"].convertToType(data['properties'], [_ArduinoProperty["default"]]);
        }

        if (data.hasOwnProperty('properties_count')) {
          obj['properties_count'] = _ApiClient["default"].convertToType(data['properties_count'], 'Number');
        }

        if (data.hasOwnProperty('sketch_id')) {
          obj['sketch_id'] = _ApiClient["default"].convertToType(data['sketch_id'], 'String');
        }

        if (data.hasOwnProperty('tags')) {
          obj['tags'] = _ApiClient["default"].convertToType(data['tags'], {
            'String': Object
          });
        }

        if (data.hasOwnProperty('timezone')) {
          obj['timezone'] = _ApiClient["default"].convertToType(data['timezone'], 'String');
        }

        if (data.hasOwnProperty('updated_at')) {
          obj['updated_at'] = _ApiClient["default"].convertToType(data['updated_at'], 'Date');
        }

        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'String');
        }

        if (data.hasOwnProperty('webhook_active')) {
          obj['webhook_active'] = _ApiClient["default"].convertToType(data['webhook_active'], 'Boolean');
        }

        if (data.hasOwnProperty('webhook_uri')) {
          obj['webhook_uri'] = _ApiClient["default"].convertToType(data['webhook_uri'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ArduinoThing;
}();
/**
 * Creation date of the thing
 * @member {Date} created_at
 */


ArduinoThing.prototype['created_at'] = undefined;
/**
 * Delete date of the thing
 * @member {Date} deleted_at
 */

ArduinoThing.prototype['deleted_at'] = undefined;
/**
 * The fqbn of the attached device, if any
 * @member {String} device_fqbn
 */

ArduinoThing.prototype['device_fqbn'] = undefined;
/**
 * The id of the device
 * @member {String} device_id
 */

ArduinoThing.prototype['device_id'] = undefined;
/**
 * The name of the attached device, if any
 * @member {String} device_name
 */

ArduinoThing.prototype['device_name'] = undefined;
/**
 * The type of the attached device, if any
 * @member {String} device_type
 */

ArduinoThing.prototype['device_type'] = undefined;
/**
 * The api reference of this thing
 * @member {String} href
 */

ArduinoThing.prototype['href'] = undefined;
/**
 * The id of the thing
 * @member {String} id
 */

ArduinoThing.prototype['id'] = undefined;
/**
 * The friendly name of the thing
 * @member {String} name
 */

ArduinoThing.prototype['name'] = undefined;
/**
 * Id of the organization the thing belongs to
 * @member {String} organization_id
 */

ArduinoThing.prototype['organization_id'] = undefined;
/**
 * ArduinoPropertyCollection is the media type for an array of ArduinoProperty (default view)
 * @member {Array.<module:model/ArduinoProperty>} properties
 */

ArduinoThing.prototype['properties'] = undefined;
/**
 * The number of properties of the thing
 * @member {Number} properties_count
 */

ArduinoThing.prototype['properties_count'] = undefined;
/**
 * The id of the attached sketch
 * @member {String} sketch_id
 */

ArduinoThing.prototype['sketch_id'] = undefined;
/**
 * Tags of the thing
 * @member {Object.<String, Object>} tags
 */

ArduinoThing.prototype['tags'] = undefined;
/**
 * Time zone of the thing
 * @member {String} timezone
 */

ArduinoThing.prototype['timezone'] = undefined;
/**
 * Update date of the thing
 * @member {Date} updated_at
 */

ArduinoThing.prototype['updated_at'] = undefined;
/**
 * The user id of the owner
 * @member {String} user_id
 */

ArduinoThing.prototype['user_id'] = undefined;
/**
 * Webhook uri
 * @member {Boolean} webhook_active
 */

ArduinoThing.prototype['webhook_active'] = undefined;
/**
 * Webhook uri
 * @member {String} webhook_uri
 */

ArduinoThing.prototype['webhook_uri'] = undefined;
var _default = ArduinoThing;
exports["default"] = _default;
