// @arduino/arduino-iot-client/dist/model/ArduinoLoradevicev1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ArduinoLoradevicev1 model module.
 * @module model/ArduinoLoradevicev1
 * @version 1.4.2
 */
var ArduinoLoradevicev1 =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>ArduinoLoradevicev1</code>.
   * ArduinoLoradevicev1 media type (default view)
   * @alias module:model/ArduinoLoradevicev1
   * @param appEui {String} The eui of the app
   * @param appKey {String} The key of the device
   * @param deviceId {String} The id of the device
   * @param eui {String} The eui of the lora device
   */
  function ArduinoLoradevicev1(appEui, appKey, deviceId, eui) {
    _classCallCheck(this, ArduinoLoradevicev1);

    ArduinoLoradevicev1.initialize(this, appEui, appKey, deviceId, eui);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ArduinoLoradevicev1, null, [{
    key: "initialize",
    value: function initialize(obj, appEui, appKey, deviceId, eui) {
      obj['app_eui'] = appEui;
      obj['app_key'] = appKey;
      obj['device_id'] = deviceId;
      obj['eui'] = eui;
    }
    /**
     * Constructs a <code>ArduinoLoradevicev1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoLoradevicev1} obj Optional instance to populate.
     * @return {module:model/ArduinoLoradevicev1} The populated <code>ArduinoLoradevicev1</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoLoradevicev1();

        if (data.hasOwnProperty('app_eui')) {
          obj['app_eui'] = _ApiClient["default"].convertToType(data['app_eui'], 'String');
        }

        if (data.hasOwnProperty('app_key')) {
          obj['app_key'] = _ApiClient["default"].convertToType(data['app_key'], 'String');
        }

        if (data.hasOwnProperty('device_id')) {
          obj['device_id'] = _ApiClient["default"].convertToType(data['device_id'], 'String');
        }

        if (data.hasOwnProperty('eui')) {
          obj['eui'] = _ApiClient["default"].convertToType(data['eui'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ArduinoLoradevicev1;
}();
/**
 * The eui of the app
 * @member {String} app_eui
 */


ArduinoLoradevicev1.prototype['app_eui'] = undefined;
/**
 * The key of the device
 * @member {String} app_key
 */

ArduinoLoradevicev1.prototype['app_key'] = undefined;
/**
 * The id of the device
 * @member {String} device_id
 */

ArduinoLoradevicev1.prototype['device_id'] = undefined;
/**
 * The eui of the lora device
 * @member {String} eui
 */

ArduinoLoradevicev1.prototype['eui'] = undefined;
var _default = ArduinoLoradevicev1;
exports["default"] = _default;
