// @arduino/arduino-iot-client/dist/model/ArduinoDevicev2propertyvalues.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ArduinoDevicev2propertyvalue = _interopRequireDefault(require("./ArduinoDevicev2propertyvalue"));

var _ArduinoDevicev2propertyvaluesLastEvaluatedKey = _interopRequireDefault(require("./ArduinoDevicev2propertyvaluesLastEvaluatedKey"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ArduinoDevicev2propertyvalues model module.
 * @module model/ArduinoDevicev2propertyvalues
 * @version 1.4.2
 */
var ArduinoDevicev2propertyvalues =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>ArduinoDevicev2propertyvalues</code>.
   * ArduinoDevicev2propertyvalues media type (default view)
   * @alias module:model/ArduinoDevicev2propertyvalues
   * @param id {String} 
   * @param lastEvaluatedKey {module:model/ArduinoDevicev2propertyvaluesLastEvaluatedKey} 
   * @param name {String} 
   * @param values {Array.<module:model/ArduinoDevicev2propertyvalue>} ArduinoDevicev2propertyvalueCollection is the media type for an array of ArduinoDevicev2propertyvalue (default view)
   */
  function ArduinoDevicev2propertyvalues(id, lastEvaluatedKey, name, values) {
    _classCallCheck(this, ArduinoDevicev2propertyvalues);

    ArduinoDevicev2propertyvalues.initialize(this, id, lastEvaluatedKey, name, values);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ArduinoDevicev2propertyvalues, null, [{
    key: "initialize",
    value: function initialize(obj, id, lastEvaluatedKey, name, values) {
      obj['id'] = id;
      obj['last_evaluated_key'] = lastEvaluatedKey;
      obj['name'] = name;
      obj['values'] = values;
    }
    /**
     * Constructs a <code>ArduinoDevicev2propertyvalues</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoDevicev2propertyvalues} obj Optional instance to populate.
     * @return {module:model/ArduinoDevicev2propertyvalues} The populated <code>ArduinoDevicev2propertyvalues</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoDevicev2propertyvalues();

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('last_evaluated_key')) {
          obj['last_evaluated_key'] = _ArduinoDevicev2propertyvaluesLastEvaluatedKey["default"].constructFromObject(data['last_evaluated_key']);
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('values')) {
          obj['values'] = _ApiClient["default"].convertToType(data['values'], [_ArduinoDevicev2propertyvalue["default"]]);
        }
      }

      return obj;
    }
  }]);

  return ArduinoDevicev2propertyvalues;
}();
/**
 * @member {String} id
 */


ArduinoDevicev2propertyvalues.prototype['id'] = undefined;
/**
 * @member {module:model/ArduinoDevicev2propertyvaluesLastEvaluatedKey} last_evaluated_key
 */

ArduinoDevicev2propertyvalues.prototype['last_evaluated_key'] = undefined;
/**
 * @member {String} name
 */

ArduinoDevicev2propertyvalues.prototype['name'] = undefined;
/**
 * ArduinoDevicev2propertyvalueCollection is the media type for an array of ArduinoDevicev2propertyvalue (default view)
 * @member {Array.<module:model/ArduinoDevicev2propertyvalue>} values
 */

ArduinoDevicev2propertyvalues.prototype['values'] = undefined;
var _default = ArduinoDevicev2propertyvalues;
exports["default"] = _default;
