// @arduino/arduino-iot-client/dist/model/ArduinoDevicev2propertyvalueValue.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ArduinoDevicev2propertyvalueValueStatistics = _interopRequireDefault(require("./ArduinoDevicev2propertyvalueValueStatistics"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ArduinoDevicev2propertyvalueValue model module.
 * @module model/ArduinoDevicev2propertyvalueValue
 * @version 1.4.2
 */
var ArduinoDevicev2propertyvalueValue =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>ArduinoDevicev2propertyvalueValue</code>.
   * @alias module:model/ArduinoDevicev2propertyvalueValue
   */
  function ArduinoDevicev2propertyvalueValue() {
    _classCallCheck(this, ArduinoDevicev2propertyvalueValue);

    ArduinoDevicev2propertyvalueValue.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ArduinoDevicev2propertyvalueValue, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ArduinoDevicev2propertyvalueValue</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoDevicev2propertyvalueValue} obj Optional instance to populate.
     * @return {module:model/ArduinoDevicev2propertyvalueValue} The populated <code>ArduinoDevicev2propertyvalueValue</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoDevicev2propertyvalueValue();

        if (data.hasOwnProperty('payload')) {
          obj['payload'] = _ApiClient["default"].convertToType(data['payload'], 'String');
        }

        if (data.hasOwnProperty('seqno')) {
          obj['seqno'] = _ApiClient["default"].convertToType(data['seqno'], 'Number');
        }

        if (data.hasOwnProperty('statistics')) {
          obj['statistics'] = _ArduinoDevicev2propertyvalueValueStatistics["default"].constructFromObject(data['statistics']);
        }
      }

      return obj;
    }
  }]);

  return ArduinoDevicev2propertyvalueValue;
}();
/**
 * @member {String} payload
 */


ArduinoDevicev2propertyvalueValue.prototype['payload'] = undefined;
/**
 * @member {Number} seqno
 */

ArduinoDevicev2propertyvalueValue.prototype['seqno'] = undefined;
/**
 * @member {module:model/ArduinoDevicev2propertyvalueValueStatistics} statistics
 */

ArduinoDevicev2propertyvalueValue.prototype['statistics'] = undefined;
var _default = ArduinoDevicev2propertyvalueValue;
exports["default"] = _default;
