// @arduino/arduino-iot-client/dist/model/ArduinoTimezone.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ArduinoTimezone model module.
 * @module model/ArduinoTimezone
 * @version 1.4.2
 */
var ArduinoTimezone =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>ArduinoTimezone</code>.
   * ArduinoTimezone media type (default view)
   * @alias module:model/ArduinoTimezone
   * @param name {String} Name of the time zone.
   * @param offset {Number} Current UTC DST offset in seconds.
   */
  function ArduinoTimezone(name, offset) {
    _classCallCheck(this, ArduinoTimezone);

    ArduinoTimezone.initialize(this, name, offset);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ArduinoTimezone, null, [{
    key: "initialize",
    value: function initialize(obj, name, offset) {
      obj['name'] = name;
      obj['offset'] = offset;
    }
    /**
     * Constructs a <code>ArduinoTimezone</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ArduinoTimezone} obj Optional instance to populate.
     * @return {module:model/ArduinoTimezone} The populated <code>ArduinoTimezone</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ArduinoTimezone();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('offset')) {
          obj['offset'] = _ApiClient["default"].convertToType(data['offset'], 'Number');
        }

        if (data.hasOwnProperty('until')) {
          obj['until'] = _ApiClient["default"].convertToType(data['until'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return ArduinoTimezone;
}();
/**
 * Name of the time zone.
 * @member {String} name
 */


ArduinoTimezone.prototype['name'] = undefined;
/**
 * Current UTC DST offset in seconds.
 * @member {Number} offset
 */

ArduinoTimezone.prototype['offset'] = undefined;
/**
 * Date until the offset is valid.
 * @member {Date} until
 */

ArduinoTimezone.prototype['until'] = undefined;
var _default = ArduinoTimezone;
exports["default"] = _default;
