// @arduino/arduino-iot-client/dist/model/BatchQueryRequestMediaV1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BatchQueryRequestMediaV1 model module.
 * @module model/BatchQueryRequestMediaV1
 * @version 1.4.2
 */
var BatchQueryRequestMediaV1 =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>BatchQueryRequestMediaV1</code>.
   * @alias module:model/BatchQueryRequestMediaV1
   * @param from {Date} From timestamp
   * @param q {String} Query
   * @param to {Date} To timestamp
   */
  function BatchQueryRequestMediaV1(from, q, to) {
    _classCallCheck(this, BatchQueryRequestMediaV1);

    BatchQueryRequestMediaV1.initialize(this, from, q, to);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BatchQueryRequestMediaV1, null, [{
    key: "initialize",
    value: function initialize(obj, from, q, to) {
      obj['from'] = from;
      obj['q'] = q;
      obj['to'] = to;
    }
    /**
     * Constructs a <code>BatchQueryRequestMediaV1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BatchQueryRequestMediaV1} obj Optional instance to populate.
     * @return {module:model/BatchQueryRequestMediaV1} The populated <code>BatchQueryRequestMediaV1</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BatchQueryRequestMediaV1();

        if (data.hasOwnProperty('from')) {
          obj['from'] = _ApiClient["default"].convertToType(data['from'], 'Date');
        }

        if (data.hasOwnProperty('interval')) {
          obj['interval'] = _ApiClient["default"].convertToType(data['interval'], 'Number');
        }

        if (data.hasOwnProperty('q')) {
          obj['q'] = _ApiClient["default"].convertToType(data['q'], 'String');
        }

        if (data.hasOwnProperty('series_limit')) {
          obj['series_limit'] = _ApiClient["default"].convertToType(data['series_limit'], 'Number');
        }

        if (data.hasOwnProperty('to')) {
          obj['to'] = _ApiClient["default"].convertToType(data['to'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return BatchQueryRequestMediaV1;
}();
/**
 * From timestamp
 * @member {Date} from
 */


BatchQueryRequestMediaV1.prototype['from'] = undefined;
/**
 * Resolution in seconds
 * @member {Number} interval
 */

BatchQueryRequestMediaV1.prototype['interval'] = undefined;
/**
 * Query
 * @member {String} q
 */

BatchQueryRequestMediaV1.prototype['q'] = undefined;
/**
 * Max of values
 * @member {Number} series_limit
 */

BatchQueryRequestMediaV1.prototype['series_limit'] = undefined;
/**
 * To timestamp
 * @member {Date} to
 */

BatchQueryRequestMediaV1.prototype['to'] = undefined;
var _default = BatchQueryRequestMediaV1;
exports["default"] = _default;
