// @arduino/arduino-iot-client/dist/model/BatchQueryRawRequestMediaV1.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The BatchQueryRawRequestMediaV1 model module.
 * @module model/BatchQueryRawRequestMediaV1
 * @version 1.4.2
 */
var BatchQueryRawRequestMediaV1 =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>BatchQueryRawRequestMediaV1</code>.
   * @alias module:model/BatchQueryRawRequestMediaV1
   * @param q {String} Query
   */
  function BatchQueryRawRequestMediaV1(q) {
    _classCallCheck(this, BatchQueryRawRequestMediaV1);

    BatchQueryRawRequestMediaV1.initialize(this, q);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(BatchQueryRawRequestMediaV1, null, [{
    key: "initialize",
    value: function initialize(obj, q) {
      obj['q'] = q;
    }
    /**
     * Constructs a <code>BatchQueryRawRequestMediaV1</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/BatchQueryRawRequestMediaV1} obj Optional instance to populate.
     * @return {module:model/BatchQueryRawRequestMediaV1} The populated <code>BatchQueryRawRequestMediaV1</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new BatchQueryRawRequestMediaV1();

        if (data.hasOwnProperty('from')) {
          obj['from'] = _ApiClient["default"].convertToType(data['from'], 'Date');
        }

        if (data.hasOwnProperty('q')) {
          obj['q'] = _ApiClient["default"].convertToType(data['q'], 'String');
        }

        if (data.hasOwnProperty('series_limit')) {
          obj['series_limit'] = _ApiClient["default"].convertToType(data['series_limit'], 'Number');
        }

        if (data.hasOwnProperty('sort')) {
          obj['sort'] = _ApiClient["default"].convertToType(data['sort'], 'String');
        }

        if (data.hasOwnProperty('to')) {
          obj['to'] = _ApiClient["default"].convertToType(data['to'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return BatchQueryRawRequestMediaV1;
}();
/**
 * From timestamp
 * @member {Date} from
 */


BatchQueryRawRequestMediaV1.prototype['from'] = undefined;
/**
 * Query
 * @member {String} q
 */

BatchQueryRawRequestMediaV1.prototype['q'] = undefined;
/**
 * Max of values
 * @member {Number} series_limit
 */

BatchQueryRawRequestMediaV1.prototype['series_limit'] = undefined;
/**
 * Sorting
 * @member {module:model/BatchQueryRawRequestMediaV1.SortEnum} sort
 * @default 'DESC'
 */

BatchQueryRawRequestMediaV1.prototype['sort'] = 'DESC';
/**
 * To timestamp
 * @member {Date} to
 */

BatchQueryRawRequestMediaV1.prototype['to'] = undefined;
/**
 * Allowed values for the <code>sort</code> property.
 * @enum {String}
 * @readonly
 */

BatchQueryRawRequestMediaV1['SortEnum'] = {
  /**
   * value: "ASC"
   * @const
   */
  "ASC": "ASC",

  /**
   * value: "DESC"
   * @const
   */
  "DESC": "DESC"
};
var _default = BatchQueryRawRequestMediaV1;
exports["default"] = _default;
