// @arduino/arduino-iot-client/dist/model/CreateDevicesV2CertsPayload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateDevicesV2CertsPayload model module.
 * @module model/CreateDevicesV2CertsPayload
 * @version 1.4.2
 */
var CreateDevicesV2CertsPayload =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>CreateDevicesV2CertsPayload</code>.
   * @alias module:model/CreateDevicesV2CertsPayload
   * @param csr {String} The certificate request in pem format
   * @param enabled {Boolean} Whether the certificate is enabled
   */
  function CreateDevicesV2CertsPayload(csr, enabled) {
    _classCallCheck(this, CreateDevicesV2CertsPayload);

    CreateDevicesV2CertsPayload.initialize(this, csr, enabled);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateDevicesV2CertsPayload, null, [{
    key: "initialize",
    value: function initialize(obj, csr, enabled) {
      obj['csr'] = csr;
      obj['enabled'] = enabled;
    }
    /**
     * Constructs a <code>CreateDevicesV2CertsPayload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateDevicesV2CertsPayload} obj Optional instance to populate.
     * @return {module:model/CreateDevicesV2CertsPayload} The populated <code>CreateDevicesV2CertsPayload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateDevicesV2CertsPayload();

        if (data.hasOwnProperty('ca')) {
          obj['ca'] = _ApiClient["default"].convertToType(data['ca'], 'String');
        }

        if (data.hasOwnProperty('csr')) {
          obj['csr'] = _ApiClient["default"].convertToType(data['csr'], 'String');
        }

        if (data.hasOwnProperty('enabled')) {
          obj['enabled'] = _ApiClient["default"].convertToType(data['enabled'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return CreateDevicesV2CertsPayload;
}();
/**
 * The Certification Authority you want to use
 * @member {String} ca
 */


CreateDevicesV2CertsPayload.prototype['ca'] = undefined;
/**
 * The certificate request in pem format
 * @member {String} csr
 */

CreateDevicesV2CertsPayload.prototype['csr'] = undefined;
/**
 * Whether the certificate is enabled
 * @member {Boolean} enabled
 */

CreateDevicesV2CertsPayload.prototype['enabled'] = undefined;
var _default = CreateDevicesV2CertsPayload;
exports["default"] = _default;
