// @arduino/arduino-iot-client/dist/model/CreateThingsV2Payload.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The CreateThingsV2Payload model module.
 * @module model/CreateThingsV2Payload
 * @version 1.3.4
 */
var CreateThingsV2Payload =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>CreateThingsV2Payload</code>.
   * ThingPayload describes a thing
   * @alias module:model/CreateThingsV2Payload
   * @param name {String} The friendly name of the thing
   */
  function CreateThingsV2Payload(name) {
    _classCallCheck(this, CreateThingsV2Payload);

    CreateThingsV2Payload.initialize(this, name);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(CreateThingsV2Payload, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }
    /**
     * Constructs a <code>CreateThingsV2Payload</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CreateThingsV2Payload} obj Optional instance to populate.
     * @return {module:model/CreateThingsV2Payload} The populated <code>CreateThingsV2Payload</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CreateThingsV2Payload();

        if (data.hasOwnProperty('device_id')) {
          obj['device_id'] = _ApiClient["default"].convertToType(data['device_id'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('webhook_active')) {
          obj['webhook_active'] = _ApiClient["default"].convertToType(data['webhook_active'], 'Boolean');
        }

        if (data.hasOwnProperty('webhook_uri')) {
          obj['webhook_uri'] = _ApiClient["default"].convertToType(data['webhook_uri'], 'String');
        }
      }

      return obj;
    }
  }]);

  return CreateThingsV2Payload;
}();
/**
 * The arn of the associated device
 * @member {String} device_id
 */


CreateThingsV2Payload.prototype['device_id'] = undefined;
/**
 * The id of the thing
 * @member {String} id
 */

CreateThingsV2Payload.prototype['id'] = undefined;
/**
 * The friendly name of the thing
 * @member {String} name
 */

CreateThingsV2Payload.prototype['name'] = undefined;
/**
 * Webhook uri
 * @member {Boolean} webhook_active
 */

CreateThingsV2Payload.prototype['webhook_active'] = undefined;
/**
 * Webhook uri
 * @member {String} webhook_uri
 */

CreateThingsV2Payload.prototype['webhook_uri'] = undefined;
var _default = CreateThingsV2Payload;
exports["default"] = _default;
