// @arduino/arduino-iot-client/dist/model/Devicev2Cert.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Devicev2Cert model module.
 * @module model/Devicev2Cert
 * @version 1.4.2
 */
var Devicev2Cert =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>Devicev2Cert</code>.
   * @alias module:model/Devicev2Cert
   */
  function Devicev2Cert() {
    _classCallCheck(this, Devicev2Cert);

    Devicev2Cert.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Devicev2Cert, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>Devicev2Cert</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Devicev2Cert} obj Optional instance to populate.
     * @return {module:model/Devicev2Cert} The populated <code>Devicev2Cert</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Devicev2Cert();

        if (data.hasOwnProperty('ca')) {
          obj['ca'] = _ApiClient["default"].convertToType(data['ca'], 'String');
        }

        if (data.hasOwnProperty('csr')) {
          obj['csr'] = _ApiClient["default"].convertToType(data['csr'], 'String');
        }

        if (data.hasOwnProperty('enabled')) {
          obj['enabled'] = _ApiClient["default"].convertToType(data['enabled'], 'Boolean');
        }
      }

      return obj;
    }
  }]);

  return Devicev2Cert;
}();
/**
 * The Certification Authority you want to use
 * @member {String} ca
 */


Devicev2Cert.prototype['ca'] = undefined;
/**
 * The certificate request in pem format
 * @member {String} csr
 */

Devicev2Cert.prototype['csr'] = undefined;
/**
 * Whether the certificate is enabled
 * @member {Boolean} enabled
 */

Devicev2Cert.prototype['enabled'] = undefined;
var _default = Devicev2Cert;
exports["default"] = _default;
