// @arduino/arduino-iot-client/dist/model/Devicev2Otabinaryurl.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The Devicev2Otabinaryurl model module.
 * @module model/Devicev2Otabinaryurl
 * @version 1.4.2
 */
var Devicev2Otabinaryurl =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>Devicev2Otabinaryurl</code>.
   * @alias module:model/Devicev2Otabinaryurl
   * @param binaryKey {String} The object key of the binary
   */
  function Devicev2Otabinaryurl(binaryKey) {
    _classCallCheck(this, Devicev2Otabinaryurl);

    Devicev2Otabinaryurl.initialize(this, binaryKey);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(Devicev2Otabinaryurl, null, [{
    key: "initialize",
    value: function initialize(obj, binaryKey) {
      obj['binary_key'] = binaryKey;
    }
    /**
     * Constructs a <code>Devicev2Otabinaryurl</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Devicev2Otabinaryurl} obj Optional instance to populate.
     * @return {module:model/Devicev2Otabinaryurl} The populated <code>Devicev2Otabinaryurl</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Devicev2Otabinaryurl();

        if (data.hasOwnProperty('async')) {
          obj['async'] = _ApiClient["default"].convertToType(data['async'], 'Boolean');
        }

        if (data.hasOwnProperty('binary_key')) {
          obj['binary_key'] = _ApiClient["default"].convertToType(data['binary_key'], 'String');
        }

        if (data.hasOwnProperty('expire_in_mins')) {
          obj['expire_in_mins'] = _ApiClient["default"].convertToType(data['expire_in_mins'], 'Number');
        }
      }

      return obj;
    }
  }]);

  return Devicev2Otabinaryurl;
}();
/**
 * If false, wait for the full OTA process, until it gets a result from the device
 * @member {Boolean} async
 * @default true
 */


Devicev2Otabinaryurl.prototype['async'] = true;
/**
 * The object key of the binary
 * @member {String} binary_key
 */

Devicev2Otabinaryurl.prototype['binary_key'] = undefined;
/**
 * Binary expire time in minutes, default 10 mins
 * @member {Number} expire_in_mins
 * @default 10
 */

Devicev2Otabinaryurl.prototype['expire_in_mins'] = 10;
var _default = Devicev2Otabinaryurl;
exports["default"] = _default;
