// @arduino/arduino-iot-client/dist/model/HistoricDataRequest.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The HistoricDataRequest model module.
 * @module model/HistoricDataRequest
 * @version 1.4.2
 */
var HistoricDataRequest =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>HistoricDataRequest</code>.
   * @alias module:model/HistoricDataRequest
   * @param from {Date} Get data starting from this date
   * @param properties {Array.<String>} IDs of properties
   * @param to {Date} Get data up to this date
   */
  function HistoricDataRequest(from, properties, to) {
    _classCallCheck(this, HistoricDataRequest);

    HistoricDataRequest.initialize(this, from, properties, to);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(HistoricDataRequest, null, [{
    key: "initialize",
    value: function initialize(obj, from, properties, to) {
      obj['from'] = from;
      obj['properties'] = properties;
      obj['to'] = to;
    }
    /**
     * Constructs a <code>HistoricDataRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/HistoricDataRequest} obj Optional instance to populate.
     * @return {module:model/HistoricDataRequest} The populated <code>HistoricDataRequest</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new HistoricDataRequest();

        if (data.hasOwnProperty('from')) {
          obj['from'] = _ApiClient["default"].convertToType(data['from'], 'Date');
        }

        if (data.hasOwnProperty('properties')) {
          obj['properties'] = _ApiClient["default"].convertToType(data['properties'], ['String']);
        }

        if (data.hasOwnProperty('to')) {
          obj['to'] = _ApiClient["default"].convertToType(data['to'], 'Date');
        }
      }

      return obj;
    }
  }]);

  return HistoricDataRequest;
}();
/**
 * Get data starting from this date
 * @member {Date} from
 */


HistoricDataRequest.prototype['from'] = undefined;
/**
 * IDs of properties
 * @member {Array.<String>} properties
 */

HistoricDataRequest.prototype['properties'] = undefined;
/**
 * Get data up to this date
 * @member {Date} to
 */

HistoricDataRequest.prototype['to'] = undefined;
var _default = HistoricDataRequest;
exports["default"] = _default;
