// @arduino/arduino-iot-client/dist/model/InlineObject.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The InlineObject model module.
 * @module model/InlineObject
 * @version 1.4.2
 */
var InlineObject =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>InlineObject</code>.
   * @alias module:model/InlineObject
   * @param otaFile {File} OTA file
   */
  function InlineObject(otaFile) {
    _classCallCheck(this, InlineObject);

    InlineObject.initialize(this, otaFile);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(InlineObject, null, [{
    key: "initialize",
    value: function initialize(obj, otaFile) {
      obj['ota_file'] = otaFile;
    }
    /**
     * Constructs a <code>InlineObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/InlineObject} obj Optional instance to populate.
     * @return {module:model/InlineObject} The populated <code>InlineObject</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InlineObject();

        if (data.hasOwnProperty('async')) {
          obj['async'] = _ApiClient["default"].convertToType(data['async'], 'Boolean');
        }

        if (data.hasOwnProperty('expire_in_mins')) {
          obj['expire_in_mins'] = _ApiClient["default"].convertToType(data['expire_in_mins'], 'Number');
        }

        if (data.hasOwnProperty('ota_file')) {
          obj['ota_file'] = _ApiClient["default"].convertToType(data['ota_file'], File);
        }
      }

      return obj;
    }
  }]);

  return InlineObject;
}();
/**
 * If false, wait for the full OTA process, until it gets a result from the device
 * @member {Boolean} async
 * @default true
 */


InlineObject.prototype['async'] = true;
/**
 * Binary expire time in minutes, default 10 mins
 * @member {Number} expire_in_mins
 * @default 10
 */

InlineObject.prototype['expire_in_mins'] = 10;
/**
 * OTA file
 * @member {File} ota_file
 */

InlineObject.prototype['ota_file'] = undefined;
var _default = InlineObject;
exports["default"] = _default;
