// @arduino/arduino-iot-client/dist/model/PropertiesValue.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The PropertiesValue model module.
 * @module model/PropertiesValue
 * @version 1.4.2
 */
var PropertiesValue =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>PropertiesValue</code>.
   * @alias module:model/PropertiesValue
   * @param name {String} The name of the property
   * @param type {module:model/PropertiesValue.TypeEnum} The type of the property
   * @param value {Object} The last value of the property
   */
  function PropertiesValue(name, type, value) {
    _classCallCheck(this, PropertiesValue);

    PropertiesValue.initialize(this, name, type, value);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(PropertiesValue, null, [{
    key: "initialize",
    value: function initialize(obj, name, type, value) {
      obj['name'] = name;
      obj['type'] = type;
      obj['value'] = value;
    }
    /**
     * Constructs a <code>PropertiesValue</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PropertiesValue} obj Optional instance to populate.
     * @return {module:model/PropertiesValue} The populated <code>PropertiesValue</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PropertiesValue();

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('type')) {
          obj['type'] = _ApiClient["default"].convertToType(data['type'], 'String');
        }

        if (data.hasOwnProperty('value')) {
          obj['value'] = _ApiClient["default"].convertToType(data['value'], Object);
        }
      }

      return obj;
    }
  }]);

  return PropertiesValue;
}();
/**
 * The name of the property
 * @member {String} name
 */


PropertiesValue.prototype['name'] = undefined;
/**
 * The type of the property
 * @member {module:model/PropertiesValue.TypeEnum} type
 * @default 'infer'
 */

PropertiesValue.prototype['type'] = 'infer';
/**
 * The last value of the property
 * @member {Object} value
 */

PropertiesValue.prototype['value'] = undefined;
/**
 * Allowed values for the <code>type</code> property.
 * @enum {String}
 * @readonly
 */

PropertiesValue['TypeEnum'] = {
  /**
   * value: "infer"
   * @const
   */
  "infer": "infer",

  /**
   * value: "base64"
   * @const
   */
  "base64": "base64",

  /**
   * value: "hex"
   * @const
   */
  "hex": "hex",

  /**
   * value: "json"
   * @const
   */
  "json": "json"
};
var _default = PropertiesValue;
exports["default"] = _default;
