// @arduino/arduino-iot-client/dist/model/ThingCreate.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _Property = _interopRequireDefault(require("./Property"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
 * The ThingCreate model module.
 * @module model/ThingCreate
 * @version 1.4.2
 */
var ThingCreate =
/*#__PURE__*/
function () {
  /**
   * Constructs a new <code>ThingCreate</code>.
   * Payload to create a new thing
   * @alias module:model/ThingCreate
   */
  function ThingCreate() {
    _classCallCheck(this, ThingCreate);

    ThingCreate.initialize(this);
  }
  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */


  _createClass(ThingCreate, null, [{
    key: "initialize",
    value: function initialize(obj) {}
    /**
     * Constructs a <code>ThingCreate</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ThingCreate} obj Optional instance to populate.
     * @return {module:model/ThingCreate} The populated <code>ThingCreate</code> instance.
     */

  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ThingCreate();

        if (data.hasOwnProperty('device_id')) {
          obj['device_id'] = _ApiClient["default"].convertToType(data['device_id'], 'String');
        }

        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'String');
        }

        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }

        if (data.hasOwnProperty('properties')) {
          obj['properties'] = _ApiClient["default"].convertToType(data['properties'], [_Property["default"]]);
        }

        if (data.hasOwnProperty('timezone')) {
          obj['timezone'] = _ApiClient["default"].convertToType(data['timezone'], 'String');
        }

        if (data.hasOwnProperty('webhook_active')) {
          obj['webhook_active'] = _ApiClient["default"].convertToType(data['webhook_active'], 'Boolean');
        }

        if (data.hasOwnProperty('webhook_uri')) {
          obj['webhook_uri'] = _ApiClient["default"].convertToType(data['webhook_uri'], 'String');
        }
      }

      return obj;
    }
  }]);

  return ThingCreate;
}();
/**
 * The arn of the associated device
 * @member {String} device_id
 */


ThingCreate.prototype['device_id'] = undefined;
/**
 * The id of the thing
 * @member {String} id
 */

ThingCreate.prototype['id'] = undefined;
/**
 * The friendly name of the thing
 * @member {String} name
 */

ThingCreate.prototype['name'] = undefined;
/**
 * The properties of the thing
 * @member {Array.<module:model/Property>} properties
 */

ThingCreate.prototype['properties'] = undefined;
/**
 * A time zone name Check /v2/timezones for a list of valid names.
 * @member {String} timezone
 * @default 'America/New_York'
 */

ThingCreate.prototype['timezone'] = 'America/New_York';
/**
 * Webhook uri
 * @member {Boolean} webhook_active
 */

ThingCreate.prototype['webhook_active'] = undefined;
/**
 * Webhook uri
 * @member {String} webhook_uri
 */

ThingCreate.prototype['webhook_uri'] = undefined;
var _default = ThingCreate;
exports["default"] = _default;
