// @arduino/arduino-iot-client/dist/api/DevicesV2TagsApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _ApiClient = _interopRequireDefault(require("../ApiClient"));

var _ArduinoTags = _interopRequireDefault(require("../model/ArduinoTags"));

var _Error = _interopRequireDefault(require("../model/Error"));

var _Tag = _interopRequireDefault(require("../model/Tag"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

/**
* DevicesV2Tags service.
* @module api/DevicesV2TagsApi
* @version 1.4.2
*/
var DevicesV2TagsApi =
/*#__PURE__*/
function () {
  /**
  * Constructs a new DevicesV2TagsApi. 
  * @alias module:api/DevicesV2TagsApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function DevicesV2TagsApi(apiClient) {
    _classCallCheck(this, DevicesV2TagsApi);

    this.apiClient = apiClient || _ApiClient["default"].instance;
  }
  /**
   * delete devices_v2_tags
   * Delete a tag associated to the device given its key.
   * @param {String} id The id of the device
   * @param {String} key The key of the tag
   * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing HTTP response
   */


  _createClass(DevicesV2TagsApi, [{
    key: "devicesV2TagsDeleteWithHttpInfo",
    value: function devicesV2TagsDeleteWithHttpInfo(id, key) {
      var postBody = null; // verify the required parameter 'id' is set

      if (id === undefined || id === null) {
        throw new _Error["default"]("Missing the required parameter 'id' when calling devicesV2TagsDelete");
      } // verify the required parameter 'key' is set


      if (key === undefined || key === null) {
        throw new _Error["default"]("Missing the required parameter 'key' when calling devicesV2TagsDelete");
      }

      var pathParams = {
        'id': id,
        'key': key
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = null;
      return this.apiClient.callApi('/v2/devices/{id}/tags/{key}', 'DELETE', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }
    /**
     * delete devices_v2_tags
     * Delete a tag associated to the device given its key.
     * @param {String} id The id of the device
     * @param {String} key The key of the tag
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}
     */

  }, {
    key: "devicesV2TagsDelete",
    value: function devicesV2TagsDelete(id, key) {
      return this.devicesV2TagsDeleteWithHttpInfo(id, key).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
    /**
     * list devices_v2_tags
     * List tags associated to the device.
     * @param {String} id The id of the device
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing data of type {@link module:model/ArduinoTags} and HTTP response
     */

  }, {
    key: "devicesV2TagsListWithHttpInfo",
    value: function devicesV2TagsListWithHttpInfo(id) {
      var postBody = null; // verify the required parameter 'id' is set

      if (id === undefined || id === null) {
        throw new _Error["default"]("Missing the required parameter 'id' when calling devicesV2TagsList");
      }

      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ArduinoTags["default"];
      return this.apiClient.callApi('/v2/devices/{id}/tags', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }
    /**
     * list devices_v2_tags
     * List tags associated to the device.
     * @param {String} id The id of the device
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with data of type {@link module:model/ArduinoTags}
     */

  }, {
    key: "devicesV2TagsList",
    value: function devicesV2TagsList(id) {
      return this.devicesV2TagsListWithHttpInfo(id).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
    /**
     * upsert devices_v2_tags
     * Creates or updates a tag associated to the device.
     * @param {String} id The id of the device
     * @param {module:model/Tag} tag 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}, with an object containing HTTP response
     */

  }, {
    key: "devicesV2TagsUpsertWithHttpInfo",
    value: function devicesV2TagsUpsertWithHttpInfo(id, tag) {
      var postBody = tag; // verify the required parameter 'id' is set

      if (id === undefined || id === null) {
        throw new _Error["default"]("Missing the required parameter 'id' when calling devicesV2TagsUpsert");
      } // verify the required parameter 'tag' is set


      if (tag === undefined || tag === null) {
        throw new _Error["default"]("Missing the required parameter 'tag' when calling devicesV2TagsUpsert");
      }

      var pathParams = {
        'id': id
      };
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['oauth2'];
      var contentTypes = ['application/json', 'application/x-www-form-urlencoded'];
      var accepts = ['application/json'];
      var returnType = null;
      return this.apiClient.callApi('/v2/devices/{id}/tags', 'PUT', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null);
    }
    /**
     * upsert devices_v2_tags
     * Creates or updates a tag associated to the device.
     * @param {String} id The id of the device
     * @param {module:model/Tag} tag 
     * @return {Promise} a {@link https://www.promisejs.org/|Promise}
     */

  }, {
    key: "devicesV2TagsUpsert",
    value: function devicesV2TagsUpsert(id, tag) {
      return this.devicesV2TagsUpsertWithHttpInfo(id, tag).then(function (response_and_data) {
        return response_and_data.data;
      });
    }
  }]);

  return DevicesV2TagsApi;
}();

exports["default"] = DevicesV2TagsApi;
