// @arduino/arduino-iot-client/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "ArduinoCompressedv2", {
  enumerable: true,
  get: function get() {
    return _ArduinoCompressedv["default"];
  }
});
Object.defineProperty(exports, "ArduinoDashboardowner", {
  enumerable: true,
  get: function get() {
    return _ArduinoDashboardowner["default"];
  }
});
Object.defineProperty(exports, "ArduinoDashboardshare", {
  enumerable: true,
  get: function get() {
    return _ArduinoDashboardshare["default"];
  }
});
Object.defineProperty(exports, "ArduinoDashboardv2", {
  enumerable: true,
  get: function get() {
    return _ArduinoDashboardv["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2Cert", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2Cert["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2EventProperties", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2EventProperties["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2Pass", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2Pass["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2SimpleProperties", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2SimpleProperties["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2Webhook", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2Webhook["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2properties", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2properties["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2propertyvalue", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2propertyvalue["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2propertyvalueValue", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2propertyvalueValue["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2propertyvalueValueStatistics", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2propertyvalueValueStatistics["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2propertyvalues", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2propertyvalues["default"];
  }
});
Object.defineProperty(exports, "ArduinoDevicev2propertyvaluesLastEvaluatedKey", {
  enumerable: true,
  get: function get() {
    return _ArduinoDevicev2propertyvaluesLastEvaluatedKey["default"];
  }
});
Object.defineProperty(exports, "ArduinoLinkedvariable", {
  enumerable: true,
  get: function get() {
    return _ArduinoLinkedvariable["default"];
  }
});
Object.defineProperty(exports, "ArduinoLoradevicev1", {
  enumerable: true,
  get: function get() {
    return _ArduinoLoradevicev["default"];
  }
});
Object.defineProperty(exports, "ArduinoLorafreqplansv1", {
  enumerable: true,
  get: function get() {
    return _ArduinoLorafreqplansv["default"];
  }
});
Object.defineProperty(exports, "ArduinoLorafreqplanv1", {
  enumerable: true,
  get: function get() {
    return _ArduinoLorafreqplanv["default"];
  }
});
Object.defineProperty(exports, "ArduinoProperty", {
  enumerable: true,
  get: function get() {
    return _ArduinoProperty["default"];
  }
});
Object.defineProperty(exports, "ArduinoSeriesBatch", {
  enumerable: true,
  get: function get() {
    return _ArduinoSeriesBatch["default"];
  }
});
Object.defineProperty(exports, "ArduinoSeriesRawBatch", {
  enumerable: true,
  get: function get() {
    return _ArduinoSeriesRawBatch["default"];
  }
});
Object.defineProperty(exports, "ArduinoSeriesRawBatchLastvalue", {
  enumerable: true,
  get: function get() {
    return _ArduinoSeriesRawBatchLastvalue["default"];
  }
});
Object.defineProperty(exports, "ArduinoSeriesRawLastValueResponse", {
  enumerable: true,
  get: function get() {
    return _ArduinoSeriesRawLastValueResponse["default"];
  }
});
Object.defineProperty(exports, "ArduinoSeriesRawResponse", {
  enumerable: true,
  get: function get() {
    return _ArduinoSeriesRawResponse["default"];
  }
});
Object.defineProperty(exports, "ArduinoSeriesResponse", {
  enumerable: true,
  get: function get() {
    return _ArduinoSeriesResponse["default"];
  }
});
Object.defineProperty(exports, "ArduinoTags", {
  enumerable: true,
  get: function get() {
    return _ArduinoTags["default"];
  }
});
Object.defineProperty(exports, "ArduinoThing", {
  enumerable: true,
  get: function get() {
    return _ArduinoThing["default"];
  }
});
Object.defineProperty(exports, "ArduinoTimeseriesmedia", {
  enumerable: true,
  get: function get() {
    return _ArduinoTimeseriesmedia["default"];
  }
});
Object.defineProperty(exports, "ArduinoTimezone", {
  enumerable: true,
  get: function get() {
    return _ArduinoTimezone["default"];
  }
});
Object.defineProperty(exports, "ArduinoVariableslinks", {
  enumerable: true,
  get: function get() {
    return _ArduinoVariableslinks["default"];
  }
});
Object.defineProperty(exports, "ArduinoWidgetv2", {
  enumerable: true,
  get: function get() {
    return _ArduinoWidgetv["default"];
  }
});
Object.defineProperty(exports, "BatchLastValueRequestsMediaV1", {
  enumerable: true,
  get: function get() {
    return _BatchLastValueRequestsMediaV["default"];
  }
});
Object.defineProperty(exports, "BatchQueryRawLastValueRequestMediaV1", {
  enumerable: true,
  get: function get() {
    return _BatchQueryRawLastValueRequestMediaV["default"];
  }
});
Object.defineProperty(exports, "BatchQueryRawRequestMediaV1", {
  enumerable: true,
  get: function get() {
    return _BatchQueryRawRequestMediaV["default"];
  }
});
Object.defineProperty(exports, "BatchQueryRawRequestsMediaV1", {
  enumerable: true,
  get: function get() {
    return _BatchQueryRawRequestsMediaV["default"];
  }
});
Object.defineProperty(exports, "BatchQueryRawResponseSeriesMediaV1", {
  enumerable: true,
  get: function get() {
    return _BatchQueryRawResponseSeriesMediaV["default"];
  }
});
Object.defineProperty(exports, "BatchQueryRequestMediaV1", {
  enumerable: true,
  get: function get() {
    return _BatchQueryRequestMediaV["default"];
  }
});
Object.defineProperty(exports, "BatchQueryRequestsMediaV1", {
  enumerable: true,
  get: function get() {
    return _BatchQueryRequestsMediaV["default"];
  }
});
Object.defineProperty(exports, "CheckDevicesV2PassPayload", {
  enumerable: true,
  get: function get() {
    return _CheckDevicesV2PassPayload["default"];
  }
});
Object.defineProperty(exports, "CreateDevicesV2CertsPayload", {
  enumerable: true,
  get: function get() {
    return _CreateDevicesV2CertsPayload["default"];
  }
});
Object.defineProperty(exports, "CreateDevicesV2Payload", {
  enumerable: true,
  get: function get() {
    return _CreateDevicesV2Payload["default"];
  }
});
Object.defineProperty(exports, "CreateLoraDevicesV1Payload", {
  enumerable: true,
  get: function get() {
    return _CreateLoraDevicesV1Payload["default"];
  }
});
Object.defineProperty(exports, "Dashboardshare", {
  enumerable: true,
  get: function get() {
    return _Dashboardshare["default"];
  }
});
Object.defineProperty(exports, "Dashboardv2", {
  enumerable: true,
  get: function get() {
    return _Dashboardv["default"];
  }
});
Object.defineProperty(exports, "Devicev2", {
  enumerable: true,
  get: function get() {
    return _Devicev["default"];
  }
});
Object.defineProperty(exports, "Devicev2Cert", {
  enumerable: true,
  get: function get() {
    return _Devicev2Cert["default"];
  }
});
Object.defineProperty(exports, "Devicev2Otabinaryurl", {
  enumerable: true,
  get: function get() {
    return _Devicev2Otabinaryurl["default"];
  }
});
Object.defineProperty(exports, "Devicev2Pass", {
  enumerable: true,
  get: function get() {
    return _Devicev2Pass["default"];
  }
});
Object.defineProperty(exports, "Error", {
  enumerable: true,
  get: function get() {
    return _Error["default"];
  }
});
Object.defineProperty(exports, "HistoricDataRequest", {
  enumerable: true,
  get: function get() {
    return _HistoricDataRequest["default"];
  }
});
Object.defineProperty(exports, "InlineObject", {
  enumerable: true,
  get: function get() {
    return _InlineObject["default"];
  }
});
Object.defineProperty(exports, "PropertiesValue", {
  enumerable: true,
  get: function get() {
    return _PropertiesValue["default"];
  }
});
Object.defineProperty(exports, "PropertiesValues", {
  enumerable: true,
  get: function get() {
    return _PropertiesValues["default"];
  }
});
Object.defineProperty(exports, "Property", {
  enumerable: true,
  get: function get() {
    return _Property["default"];
  }
});
Object.defineProperty(exports, "PropertyValue", {
  enumerable: true,
  get: function get() {
    return _PropertyValue["default"];
  }
});
Object.defineProperty(exports, "Sharerequest", {
  enumerable: true,
  get: function get() {
    return _Sharerequest["default"];
  }
});
Object.defineProperty(exports, "Tag", {
  enumerable: true,
  get: function get() {
    return _Tag["default"];
  }
});
Object.defineProperty(exports, "ThingCreate", {
  enumerable: true,
  get: function get() {
    return _ThingCreate["default"];
  }
});
Object.defineProperty(exports, "ThingSketch", {
  enumerable: true,
  get: function get() {
    return _ThingSketch["default"];
  }
});
Object.defineProperty(exports, "ThingUpdate", {
  enumerable: true,
  get: function get() {
    return _ThingUpdate["default"];
  }
});
Object.defineProperty(exports, "TimeseriesDataPoint", {
  enumerable: true,
  get: function get() {
    return _TimeseriesDataPoint["default"];
  }
});
Object.defineProperty(exports, "UpdateSketch", {
  enumerable: true,
  get: function get() {
    return _UpdateSketch["default"];
  }
});
Object.defineProperty(exports, "Widget", {
  enumerable: true,
  get: function get() {
    return _Widget["default"];
  }
});
Object.defineProperty(exports, "Widgetlink", {
  enumerable: true,
  get: function get() {
    return _Widgetlink["default"];
  }
});
Object.defineProperty(exports, "DashboardsV2Api", {
  enumerable: true,
  get: function get() {
    return _DashboardsV2Api["default"];
  }
});
Object.defineProperty(exports, "DevicesV2Api", {
  enumerable: true,
  get: function get() {
    return _DevicesV2Api["default"];
  }
});
Object.defineProperty(exports, "DevicesV2CertsApi", {
  enumerable: true,
  get: function get() {
    return _DevicesV2CertsApi["default"];
  }
});
Object.defineProperty(exports, "DevicesV2OtaApi", {
  enumerable: true,
  get: function get() {
    return _DevicesV2OtaApi["default"];
  }
});
Object.defineProperty(exports, "DevicesV2PassApi", {
  enumerable: true,
  get: function get() {
    return _DevicesV2PassApi["default"];
  }
});
Object.defineProperty(exports, "DevicesV2TagsApi", {
  enumerable: true,
  get: function get() {
    return _DevicesV2TagsApi["default"];
  }
});
Object.defineProperty(exports, "LoraDevicesV1Api", {
  enumerable: true,
  get: function get() {
    return _LoraDevicesV1Api["default"];
  }
});
Object.defineProperty(exports, "LoraFreqPlanV1Api", {
  enumerable: true,
  get: function get() {
    return _LoraFreqPlanV1Api["default"];
  }
});
Object.defineProperty(exports, "PropertiesV2Api", {
  enumerable: true,
  get: function get() {
    return _PropertiesV2Api["default"];
  }
});
Object.defineProperty(exports, "SeriesV2Api", {
  enumerable: true,
  get: function get() {
    return _SeriesV2Api["default"];
  }
});
Object.defineProperty(exports, "ThingsV2Api", {
  enumerable: true,
  get: function get() {
    return _ThingsV2Api["default"];
  }
});
Object.defineProperty(exports, "ThingsV2TagsApi", {
  enumerable: true,
  get: function get() {
    return _ThingsV2TagsApi["default"];
  }
});

var _ApiClient = _interopRequireDefault(require("./ApiClient"));

var _ArduinoCompressedv = _interopRequireDefault(require("./model/ArduinoCompressedv2"));

var _ArduinoDashboardowner = _interopRequireDefault(require("./model/ArduinoDashboardowner"));

var _ArduinoDashboardshare = _interopRequireDefault(require("./model/ArduinoDashboardshare"));

var _ArduinoDashboardv = _interopRequireDefault(require("./model/ArduinoDashboardv2"));

var _ArduinoDevicev = _interopRequireDefault(require("./model/ArduinoDevicev2"));

var _ArduinoDevicev2Cert = _interopRequireDefault(require("./model/ArduinoDevicev2Cert"));

var _ArduinoDevicev2EventProperties = _interopRequireDefault(require("./model/ArduinoDevicev2EventProperties"));

var _ArduinoDevicev2Pass = _interopRequireDefault(require("./model/ArduinoDevicev2Pass"));

var _ArduinoDevicev2SimpleProperties = _interopRequireDefault(require("./model/ArduinoDevicev2SimpleProperties"));

var _ArduinoDevicev2Webhook = _interopRequireDefault(require("./model/ArduinoDevicev2Webhook"));

var _ArduinoDevicev2properties = _interopRequireDefault(require("./model/ArduinoDevicev2properties"));

var _ArduinoDevicev2propertyvalue = _interopRequireDefault(require("./model/ArduinoDevicev2propertyvalue"));

var _ArduinoDevicev2propertyvalueValue = _interopRequireDefault(require("./model/ArduinoDevicev2propertyvalueValue"));

var _ArduinoDevicev2propertyvalueValueStatistics = _interopRequireDefault(require("./model/ArduinoDevicev2propertyvalueValueStatistics"));

var _ArduinoDevicev2propertyvalues = _interopRequireDefault(require("./model/ArduinoDevicev2propertyvalues"));

var _ArduinoDevicev2propertyvaluesLastEvaluatedKey = _interopRequireDefault(require("./model/ArduinoDevicev2propertyvaluesLastEvaluatedKey"));

var _ArduinoLinkedvariable = _interopRequireDefault(require("./model/ArduinoLinkedvariable"));

var _ArduinoLoradevicev = _interopRequireDefault(require("./model/ArduinoLoradevicev1"));

var _ArduinoLorafreqplansv = _interopRequireDefault(require("./model/ArduinoLorafreqplansv1"));

var _ArduinoLorafreqplanv = _interopRequireDefault(require("./model/ArduinoLorafreqplanv1"));

var _ArduinoProperty = _interopRequireDefault(require("./model/ArduinoProperty"));

var _ArduinoSeriesBatch = _interopRequireDefault(require("./model/ArduinoSeriesBatch"));

var _ArduinoSeriesRawBatch = _interopRequireDefault(require("./model/ArduinoSeriesRawBatch"));

var _ArduinoSeriesRawBatchLastvalue = _interopRequireDefault(require("./model/ArduinoSeriesRawBatchLastvalue"));

var _ArduinoSeriesRawLastValueResponse = _interopRequireDefault(require("./model/ArduinoSeriesRawLastValueResponse"));

var _ArduinoSeriesRawResponse = _interopRequireDefault(require("./model/ArduinoSeriesRawResponse"));

var _ArduinoSeriesResponse = _interopRequireDefault(require("./model/ArduinoSeriesResponse"));

var _ArduinoTags = _interopRequireDefault(require("./model/ArduinoTags"));

var _ArduinoThing = _interopRequireDefault(require("./model/ArduinoThing"));

var _ArduinoTimeseriesmedia = _interopRequireDefault(require("./model/ArduinoTimeseriesmedia"));

var _ArduinoTimezone = _interopRequireDefault(require("./model/ArduinoTimezone"));

var _ArduinoVariableslinks = _interopRequireDefault(require("./model/ArduinoVariableslinks"));

var _ArduinoWidgetv = _interopRequireDefault(require("./model/ArduinoWidgetv2"));

var _BatchLastValueRequestsMediaV = _interopRequireDefault(require("./model/BatchLastValueRequestsMediaV1"));

var _BatchQueryRawLastValueRequestMediaV = _interopRequireDefault(require("./model/BatchQueryRawLastValueRequestMediaV1"));

var _BatchQueryRawRequestMediaV = _interopRequireDefault(require("./model/BatchQueryRawRequestMediaV1"));

var _BatchQueryRawRequestsMediaV = _interopRequireDefault(require("./model/BatchQueryRawRequestsMediaV1"));

var _BatchQueryRawResponseSeriesMediaV = _interopRequireDefault(require("./model/BatchQueryRawResponseSeriesMediaV1"));

var _BatchQueryRequestMediaV = _interopRequireDefault(require("./model/BatchQueryRequestMediaV1"));

var _BatchQueryRequestsMediaV = _interopRequireDefault(require("./model/BatchQueryRequestsMediaV1"));

var _CheckDevicesV2PassPayload = _interopRequireDefault(require("./model/CheckDevicesV2PassPayload"));

var _CreateDevicesV2CertsPayload = _interopRequireDefault(require("./model/CreateDevicesV2CertsPayload"));

var _CreateDevicesV2Payload = _interopRequireDefault(require("./model/CreateDevicesV2Payload"));

var _CreateLoraDevicesV1Payload = _interopRequireDefault(require("./model/CreateLoraDevicesV1Payload"));

var _Dashboardshare = _interopRequireDefault(require("./model/Dashboardshare"));

var _Dashboardv = _interopRequireDefault(require("./model/Dashboardv2"));

var _Devicev = _interopRequireDefault(require("./model/Devicev2"));

var _Devicev2Cert = _interopRequireDefault(require("./model/Devicev2Cert"));

var _Devicev2Otabinaryurl = _interopRequireDefault(require("./model/Devicev2Otabinaryurl"));

var _Devicev2Pass = _interopRequireDefault(require("./model/Devicev2Pass"));

var _Error = _interopRequireDefault(require("./model/Error"));

var _HistoricDataRequest = _interopRequireDefault(require("./model/HistoricDataRequest"));

var _InlineObject = _interopRequireDefault(require("./model/InlineObject"));

var _PropertiesValue = _interopRequireDefault(require("./model/PropertiesValue"));

var _PropertiesValues = _interopRequireDefault(require("./model/PropertiesValues"));

var _Property = _interopRequireDefault(require("./model/Property"));

var _PropertyValue = _interopRequireDefault(require("./model/PropertyValue"));

var _Sharerequest = _interopRequireDefault(require("./model/Sharerequest"));

var _Tag = _interopRequireDefault(require("./model/Tag"));

var _ThingCreate = _interopRequireDefault(require("./model/ThingCreate"));

var _ThingSketch = _interopRequireDefault(require("./model/ThingSketch"));

var _ThingUpdate = _interopRequireDefault(require("./model/ThingUpdate"));

var _TimeseriesDataPoint = _interopRequireDefault(require("./model/TimeseriesDataPoint"));

var _UpdateSketch = _interopRequireDefault(require("./model/UpdateSketch"));

var _Widget = _interopRequireDefault(require("./model/Widget"));

var _Widgetlink = _interopRequireDefault(require("./model/Widgetlink"));

var _DashboardsV2Api = _interopRequireDefault(require("./api/DashboardsV2Api"));

var _DevicesV2Api = _interopRequireDefault(require("./api/DevicesV2Api"));

var _DevicesV2CertsApi = _interopRequireDefault(require("./api/DevicesV2CertsApi"));

var _DevicesV2OtaApi = _interopRequireDefault(require("./api/DevicesV2OtaApi"));

var _DevicesV2PassApi = _interopRequireDefault(require("./api/DevicesV2PassApi"));

var _DevicesV2TagsApi = _interopRequireDefault(require("./api/DevicesV2TagsApi"));

var _LoraDevicesV1Api = _interopRequireDefault(require("./api/LoraDevicesV1Api"));

var _LoraFreqPlanV1Api = _interopRequireDefault(require("./api/LoraFreqPlanV1Api"));

var _PropertiesV2Api = _interopRequireDefault(require("./api/PropertiesV2Api"));

var _SeriesV2Api = _interopRequireDefault(require("./api/SeriesV2Api"));

var _ThingsV2Api = _interopRequireDefault(require("./api/ThingsV2Api"));

var _ThingsV2TagsApi = _interopRequireDefault(require("./api/ThingsV2TagsApi"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
